From ff887e741a5d37f59a1d3b5d27f9f3ea16f8e978 Mon Sep 17 00:00:00 2001
From: Guoqing Jiang <guoqing.jiang@suse.com>
Date: Mon, 28 Mar 2022 14:19:35 +0800
Subject: [PATCH] Revert "dmaengine: idxd: Fix potential null dereference on
 pointer status"
Patch-mainline: Never, revert stable patch which is added again in sorted section
References: bsn#161
Modified-by-SEL: No


This reverts commit 5756f757c72501ef1a16f5f63f940623044180e9.

Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/dma/idxd/device.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/dma/idxd/device.c b/drivers/dma/idxd/device.c
index f90fdf7aeeef..459e9fbc2253 100644
--- a/drivers/dma/idxd/device.c
+++ b/drivers/dma/idxd/device.c
@@ -378,8 +378,7 @@ static void idxd_cmd_exec(struct idxd_device *idxd, int cmd_code, u32 operand,
 
 	if (idxd_device_is_halted(idxd)) {
 		dev_warn(&idxd->pdev->dev, "Device is HALTED!\n");
-		if (status)
-			*status = IDXD_CMDSTS_HW_ERR;
+		*status = IDXD_CMDSTS_HW_ERR;
 		return;
 	}
 
-- 
2.26.2

